src/OrchardCore.Modules/OrchardCore.Setup/Startup.cs

```diff
@@ -1,20 +1,60 @@
 using System;
+using System.Linq;
 using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Hosting;
 using Microsoft.AspNetCore.Routing;
+using Microsoft.Extensions.Configuration;
 using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Options;
 using OrchardCore.Modules;
 
 namespace OrchardCore.Setup
 {
     public class Startup : StartupBase
     {
+        public Startup(IHostingEnvironment env)
+        {
+            var builder = new ConfigurationBuilder()
+                .SetBasePath(env.ContentRootPath)
+                .AddJsonFile("appsettings.json");
+
+            Configuration = builder.Build();
+        }
+
+        public IConfiguration Configuration { get; }
+
         public override void ConfigureServices(IServiceCollection services)
         {
+            services.AddPortableObjectLocalization(options => options.ResourcesPath = "Localization");
             services.AddSetup();
-		}
+        }
 
-		public override void Configure(IApplicationBuilder builder, IRouteBuilder routes, IServiceProvider serviceProvider)
+		public override void Configure(IApplicationBuilder app, IRouteBuilder routes, IServiceProvider serviceProvider)
         {
+            var options = serviceProvider.GetService<IOptions<RequestLocalizationOptions>>().Value;
+            var defaultCulture = Configuration["OrchardCore:OrchardCore.Setup:DefaultCulture"];
```

Comment:
So, without the `OrchardCore` section and maybe a null check if it is intended to use the invariant culture if the default culture is an empty string in the `appsettings.json`. Anyway, below we check `IsNullOrEmpty`, so as you want.

    var defaultCulture = Configuration["OrchardCore.Setup:DefaultCulture"] ?? String.Empty;

and then

    var supportedCultures = Configuration.GetSection("OrchardCore.Setup:SupportedCultures").Get<string[]>();